sdk/identity/azure-identity/src/main/java/com/azure/identity/package-info.java

```diff
@@ -2,7 +2,277 @@
+ *
+ * <h2>Getting Started</h2>
+ *
+ * <p>The {@link com.azure.identity.DefaultAzureCredential} is appropriate for most scenarios where the application is
+ * intended to ultimately be run in Azure. This is because the {@link com.azure.identity.DefaultAzureCredential}
+ * combines credentials commonly used to authenticate when deployed, with credentials used to authenticate in a
+ * development environment. Note, this credential is intended to simplify getting started with the SDK by handling
+ * common scenarios with reasonable default behaviors. Developers who want more control or whose scenario isn't
+ * served by the default settings should use other credential types. For more information refer to the
+ * <a href="https://aka.ms/azsdk/java/identity/defaultazurecredential/docs">conceptual knowledge and configuration
+ * details</a>.</p>
+ *
+ * <p><strong>Sample: Construct a simple DefaultAzureCredential</strong></p>
+ *
+ * <!-- src_embed com.azure.identity.credential.defaultazurecredential.construct -->
+ * <pre>
+ * TokenCredential defaultAzureCredential = new DefaultAzureCredentialBuilder&#40;&#41;
+ *     .build&#40;&#41;;
+ * </pre>
+ * <!-- end com.azure.identity.credential.defaultazurecredential.construct -->
+ *
+ * <p>The Azure SDK client builders consume TokenCredential for Azure Active Directory (AAD) based authentication.
+ * The TokenCredential instantiated above can be passed into most of the Azure SDK client builders for
+ * AAD authentication.</p>
+ *
+ * <p>The {@link com.azure.identity.DefaultAzureCredential} works well in most of the scenarios as it executes a chain
+ * of credentials underneath which covers well known authentication scenarios for both Azure hosted platforms and
+ * development environment. But, in some scenarios where only a specific authentication mechanism will work, it is
+ * recommended to use that specific credential to authenticate. Let's take a look at the individual
+ * authentication scenarios and their respective credential use below.</p>
+ *
+ * <hr/>
+ *
+ * <h2> Authenticating on Azure Hosted Platforms via Managed Identity</h2>
+ *
+ * <p>The {@link com.azure.identity.ManagedIdentityCredential} authenticates the configured managed identity
+ * (system or user assigned) of an Azure resource. So, if the application is running inside an Azure resource that
+ * supports Managed Identity through IDENTITY/MSI, IMDS endpoints, or both, then the
+ * {@link com.azure.identity.ManagedIdentityCredential} will get your application authenticated, and offers a great
+ * secretless authentication experience. For more information refer to the
+ * <a href="https://aka.ms/azsdk/java/identity/managedidentity/docs">conceptual knowledge and configuration
+ * details</a>.</p>
+ *
+ * <p><strong>Sample: Construct a Managed Identity Credential</strong></p>
+ *
+ * <!-- src_embed com.azure.identity.credential.managedidentitycredential.construct -->
```

Comment:
Same comment here about potentially introducing a short description - especially if you want to encourage the user to read the class-level JavaDoc for more configuration options, etc.